### PR TITLE
feat(dropdowns): update react-dropdowns to align with next example and prop naming

### PR DESCRIPTION
**Pull request by austin94**
- [x] **BREAKING CHANGE**

## Description

This PR updates the directory structure and prop naming within `react-dropdowns` to match the other packages in the `next` branch.

This PR doesn't attempt to update _all_ props as these will align with the other packages during the CSS-in-JS follow-on PR. (i.e. `Label`, `Hint`, `Message` will eventually align with `react-forms`).

I am not attempting to fully update the examples in this PR.

## Detail

This was the first TypeScript package in `react-components` and there was some invalid usages of `HTMLProps`. I've moved these over to `HTMLAttributes` throughout `dropdowns` and `datepickers`.

## Current Breaking Changes

- The following boolean props have been renamed:
  - `Autocomplete`, `Multiselect`, `Select`
    - `small` -> `isCompact`
    - `isBare` -> `isBare`
    - `focused` -> `isFocused`
    - `hovered` -> `isHovered`
    - `open` -> `isOpen`
  - `Hint`
    - `small` -> `isSmall`
  - `Label`
    - `regular` -> `isRegular`
    - `small` -> `isCompact`
  - `AddItem`, `HeaderItem`, `Item`, `MediaItem`, `NextItem`, `PreviousItem`
    - `active` -> `isActive`
    - `focused` -> `isFocused`
    - `hovered` -> `isHovered`
  - `Menu`
    - `animate` -> `isAnimated`
    - `small` -> `isCompact`
    - `hidden` -> `isHidden`
    - `arrow` -> `hasArrow`

## Checklist

- [ ] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [x] :nail_care: view component styling is based on a Garden CSS
      component
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by jzempel** [comment]
So do you intend to migrate `isSmall` props to `isCompact` in a future PR?

**Comment by austin94** [comment]
> So do you intend to migrate `isSmall` props to `isCompact` in a future PR?

My original intention was to wait until I convert over the `react-forms` related styling, but I neglected the `small` prop on `Menu`. I've gone ahead and converted all `isSmall` props to `isCompact`.

File: packages/tags/src/styled/StyledTag.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ const colorStyles = (props: IStyledTagProps & ThemeProps<DefaultTheme>) => {
       color: ${foregroundColor}; /* <a> element reset */
     }
 
-    &[data-garden-focus-visible] {
+    &:focus {
```
[the remark this answers is not in the document]
Comment (austin94) [reply packages/tags/src/styled/StyledTag.ts]: Aw, forgot to leave a PR comment explaining this.

With Tags we use roving tab-index to denote selection. For all selectable tag usages (`[tabIndex='0']) we show the focus ring regardless of whether it was selected via a keyboard or a mouse.

Without this change the tags do not show a focus ring when selected via mouse.

File: BREAKING_CHANGES.md [hunk 2]
```diff
@@ -71,6 +71,30 @@ should be re-checked for ID naming accuracy.
...
+  - `Label`
+    - `regular` -> `isRegular`
+    - `small` -> `isSmall`
+  - `AddItem`, `HeaderItem`, `Item`, `MediaItem`, `NextItem`, `PreviousItem`
+    - `active` -> `isActive`
+    - `focused` -> `isFocused`
+    - `hovered` -> `isHovered`
+  - `Menu`
+    - `animate` -> `isAnimated`
+    - `small` -> `isSmall`
```
[the remark this answers is not in the document]
Comment (austin94) [reply BREAKING_CHANGES.md]: Updated.

File: packages/tags/src/styled/StyledTag.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/tags/src/styled/StyledTag.ts]: Ah, this makes sense. Although it's kind of awkward to bury it under this PR.

[the remark this answers is not in the document]
Comment (jzempel) [reply packages/tags/src/styled/StyledTag.ts]: In addition, an inline comment would help explain why this treatment is different from other components.

[the remark this answers is not in the document]
Comment (austin94) [reply packages/tags/src/styled/StyledTag.ts]: I've added an inline comment to help explain the difference.

**Pull request merged**